doc/groups.md

```diff
@@ -1 +1,519 @@
+        "modified": "2020-12-01 00:00:00"
+    }
+} 
+```
+### Logic
+#### Endpoint Validation
+- must be add parameter id
+- id must exist in database
+- must owned by user login
+
+#### Validation
+No Validation
+
+### Scenario Test
+
+#### Case : Negative Case 1
+
+Groups is not owned by user login
+
+Response HTTP Status Code : 403
+
+Response Payload :
+```
+{
+    "status_code": "CDC-403",
+    "status_message": "Groups is not owned by user login",
+    "data": null
+}
+```
+
+#### Case : Negative Case 2
+
+- param id is not exist in database
+
+Endpoint : /groups/100
+
+Response HTTP Status Code : 404
+
+Response Payload :
+```
+{
+    "status_code": "CDC-404",
+    "status_message": "data not found",
+    "data": null
+}
+```
+
+#### Case : Positive Case 1
+
+- param id is exist in database
+
+Endpoint : /groups/1
+
+Headers :
+Key | Value 
+--- | ---
+Content-Type | application/json
+Accept | application/json
+Email | administrator@gmail.com
+
+Request Payload :
+```
+{
+   "title": "myAdmin",
+   "description": "My Administrator"
+}
+```
+
+Response HTTP Status Code : 200
+
+Response Payload :
+```
+{
+    "status_code": "CDC-200",
+    "status_message": "Data Changed",
+    "data": {
+        "id": "1",
+        "title": "myAdmin",
+        "description": "Administrator",
+        "created": "2020-12-01 00:00:00",
+        "modified": "2020-12-11 12:00:00"
+    }
+}
+```
+
+#### Case : Positive Case 2
+
+Request Payload : empty
+
+Endpoint : /groups/1
+
+Response HTTP Status Code : 200
+
+Response Payload :
+```
+{
+    "status_code": "CDC-200",
+    "status_message": "No data was changed",
+    "data": {
+        "id": "1",
+        "title": "myAdmin",
+        "description": "Administrator",
+        "created": "2020-12-01 00:00:00",
+        "modified": "2020-12-11 12:00:00"
+    }
+}
+```
+
+#### Case : Positive Case 3
+
+Request Payload :
+```
+{}
+```
+Endpoint : /groups/1
+
+Response HTTP Status Code : 200
+
+Response Payload :
+```
+{
+    "status_code": "CDC-200",
+    "status_message": "No data was changed",
+    "data": {
+        "id": "1",
+        "title": "myAdmin",
+        "description": "Administrator",
+        "created": "2020-12-01 00:00:00",
+        "modified": "2020-12-11 12:00:00"
+    }
+}
+```
+
+## <a name="view"></a>View All Groups
+
+### Endpoint
+GET /groups
+
+### Headers
+Key | Value 
+--- | ---
+Content-Type | application/json
+Accept | application/json
+Email | administrator@mail.com
+
+### Request Payloads
+Name | Type | Example Value
+--- | --- | ---
+title | string | Admin  
+description | string | Administrator
+
+
+### Request Payload
+
+No Request Payload
+
+### Response Payloads
+HTTP Code | Status | Description
+--- | --- | ---
+400 | Bad Request | Bad request payload  
+404 | Not Found | User not found in database  
+500 | Internal Server Error | some un-handle error in server 
+200 | OK | OK
+```
+{
+    "status_code": "CDC-400",
+    "status_message": "Bad Request",
```

Comment:
"status_message": "Internal Server Error",

Reply:
there is no bad request on GET /groups
please change with 500 internal server error for example of error response